models/template.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const Joi = require('joi');
+const mutate = require('../lib/mutate');
+const Template = require('../config/template');
+const scmUri = Joi.reach(require('./pipeline').base, 'scmUri');
+
+const MODEL = {
+    id: Joi
+        .number().integer().positive()
+        .description('Identifier of this template')
+        .example(123345),
+
+    labels: Joi
+        .array()
+        .items(Joi.string())
+        .description('Labels for template')
+        .example(['stable', 'latest', 'beta']),
+
+    config: Joi.reach(Template.template, 'config'),
```

Comment:
Maybe we can just expose those fields in `../config/template`, so you won't need to do `Joi.reach` here multiple times